dynamo/tools/clustering.py

```diff
@@ -174,36 +173,61 @@ def hdbscan(
+    adj_matrix: Optional[csr_matrix] = None,
+    adj_matrix_key: Optional[str] = None,
+    result_key: Optional[str] = None,
+    layer: Optional[str] = None,
+    obsm_key: Optional[str] = None,
+    selected_cluster_subset: Optional[list] = None,
+    selected_cell_subset: Optional[list] = None,
+    directed: bool = False,
+    copy: bool = False,
     **kwargs
 ) -> anndata.AnnData:
     """Apply leiden clustering to adata.
-    For other community detection general parameters, please refer to ``dynamo's`` :py:meth:`~dynamo.tl.cluster_community` function.
-    "The Leiden algorithm is an improvement of the Louvain algorithm. The Leiden algorithm consists of three phases: (1) local moving of nodes, (2) refinement of the partition (3) aggregation of the network based on the refined partition, using the non-refined partition to create an initial partition for the aggregate network." - cdlib
+    For other community detection general parameters,
+    please refer to ``dynamo's`` :py:meth:`~dynamo.tl.cluster_community` function.
+    "The Leiden algorithm is an improvement of the Louvain algorithm.
+    The Leiden algorithm consists of three phases:
+    (1) local moving of nodes,
+    (2) refinement of the partition,
+    (3) aggregation of the network based on the refined partition,
+    using the non-refined partition to create an initial partition for the aggregate network." - cdlib
 
-    Parameters
-    ----------
-    weight :
-         weights of edges. Can be either an iterable or an edge attribute. Default None
-    initial_membership : optional
-        list of int Initial membership for the partition. If None then defaults to a singleton partition. Default None, by default None
+    Args:
+        adata: an adata object
+        resolution: the resolution of the clustering that determines the level of detail in the clustering process.
+            An increase in this value will result in the generation of a greater number of clusters.
+        use_weight: whether to use the weight of the edges in the clustering process. Default True
+        weight: weights of edges. Can be either an iterable(list of double) or an edge attribute. Default None
+        initial_membership: list of int Initial membership for the partition.
+            If None then defaults to a singleton partition.
+        adj_matrix: the adjacency matrix to use for clustering. Default None
+        adj_matrix_key: adj_matrix_key in adata.obsp used for clustering. Default None
+        result_key: the key to use for the clustering results. Default None
+        layer: the adata layer where cluster algorithms will work on, by default None
+        obsm_key: the key of the obsm to use for a function of neighbors. Default None
+        selected_cluster_subset: a tuple of 2 elements (cluster_key, allowed_clusters).
+            filtering cells in adata based on cluster_key in adata.obs and only reserve cells in the allowed clusters.
```

Comment:
Glad you are breaking lines, but make sure you don't break too much. the code should more or less right aligned. pre-commit should be able to automatically break lines for you. 

If not, figure out what is wrong in your end